components/system-broker/tests/common/test_context.go

```diff
@@ -0,0 +1,167 @@
+package common
+
+import (
+	"context"
+	"github.com/gavv/httpexpect"
+	"github.com/kyma-incubator/compass/components/system-broker/internal/config"
+	"github.com/kyma-incubator/compass/components/system-broker/pkg/env"
+	"github.com/kyma-incubator/compass/components/system-broker/pkg/server"
+	"github.com/kyma-incubator/compass/components/system-broker/pkg/uuid"
+	"math"
+	"math/rand"
+	"net"
+	"net/http"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+)
+
+const SystemBrokerServer = "system-broker-server"
+
+type TestContext struct {
+	SystemBroker *httpexpect.Expect
+
+	servers map[string]FakeServer
+}
+
+func (tc *TestContext) CleanUp() {
+	for _, server := range tc.servers {
+		server.Close()
+	}
+}
+
+type TestContextBuilder struct {
+	envHooks []func(env env.Environment, servers map[string]FakeServer)
+
+	Environment env.Environment
+
+	Servers    map[string]FakeServer
+	HttpClient *http.Client
+}
+
+// NewTestContextBuilder sets up a builder with default values
+func NewTestContextBuilder() *TestContextBuilder {
+	return &TestContextBuilder{
+		Environment: TestEnv(),
```

Comment:
I was wondering where we get the default env values from, as far as I remember back in SM, we had a dedicated `application.yml` for default env values. But I see here that we don't really need anything more than the values that you override in the envHooks + we rely on the `DefaultConfig()` values.

Reply:
Yes here we have a self contained default config which is sufficient for now. If we later introduce a more sophisticated tests which need external configuration we can add it but since we started lean with the component tests framework this feature is still not needed.